src/util/messages.ml

```diff
@@ -149,19 +293,10 @@ let warn_each ?ctx msg =
       end
   end
 
-(*
-let warn_each_ctx ctx msg = (* cyclic dependency... *)
-  if not @@ GobConfig.get_bool "dbg.warn_with_context" then warn_each msg else
-  (* let module S = (val Control.get_spec ()) in *)
-  (* warn_each (msg ^ " in context " ^ S.C.short 99999 (Obj.obj ctx.context ())) *)
-  (* warn_each (msg ^ " in context " ^ IO.to_string S.C.printXml (Obj.obj ctx.context ())) *)
-  warn_each (msg ^ " in context " ^ string_of_int (Hashtbl.hash (Obj.obj ctx.context ())))
-*)
-
 let debug msg =
-  if (get_bool "dbg.debug") then warn ("{BLUE}"^msg)
+  if (get_bool "dbg.debug") then warn (Warning.debug ("{BLUE}"^msg))
```

Comment:
We discussed this yesterday in the meeting. I suggest you leave the `debug` and `debug_each` as you have implemented it right now.